task/sast-shell-check/0.1/sast-shell-check.yaml

```diff
@@ -0,0 +1,179 @@
+
+        OUTPUT_FILE="shellcheck-results.json"
+        SOURCE_CODE_DIR=$(workspaces.workspace.path)/source
+
+        # generate all shellcheck result JSON files to $SC_RESULTS_DIR, which defaults to ./shellcheck-results/
+        /usr/share/csmock/scripts/run-shellcheck.sh "$SOURCE_CODE_DIR"
+
+        CSGREP_OPTS=(
+            --mode=json
+            --strip-path-prefix="$SOURCE_CODE_DIR"/
+            --event='error|warning'
+            --remove-duplicates
+            --embed-context=3
+        )
+        csgrep "${CSGREP_OPTS[@]}" ./shellcheck-results/*.json > "$OUTPUT_FILE"
+
+        if [ $? -ne 0 ]; then
+            echo "Error occurred while running 'run-shellcheck.sh'"
+            note="Task $(context.task.name) failed: For details, check Tekton task log."
+            ERROR_OUTPUT=$(make_result_json -r ERROR -t "$note")
+            echo "${ERROR_OUTPUT}" | tee "$(results.TEST_OUTPUT.path)"
+            exit 1
+        fi
+
+        # Filter known false positives if KFP_GIT_URL is set
+        if [ -n "${KFP_GIT_URL}" ]; then
+            echo "Filtering known false positives using ${KFP_GIT_URL}"
+
+            # build initial csfilter-kfp command
+            csfilter_cmd="csfilter-kfp --verbose --kfp-git-url=${KFP_GIT_URL}"
+
+            # Append --project-nvr option if PROJECT_NVR is set
+            if [ -n "${PROJECT_NVR}" ]; then
+                csfilter_cmd+=" --project-nvr=${PROJECT_NVR}"
+            fi
+
+            if [[ "${RECORD_EXCLUDED}" == "true" ]]; then
+                csfilter_cmd+=" --record-excluded=excluded-findings.json"
+            fi
+
+            # Execute the command and capture any errors
+            if ! "${csfilter_cmd[@]}" "$OUTPUT_FILE" > "${OUTPUT_FILE}.filtered" 2> "${OUTPUT_FILE}.error"; then
+                echo "Error occurred while filtering known false positives:"
+                cat "${OUTPUT_FILE}.error"
+                note="Task $(context.task.name) failed: For details, check Tekton task log."
+                ERROR_OUTPUT=$(make_result_json -r ERROR -t "$note")
+                echo "${ERROR_OUTPUT}" | tee "$(results.TEST_OUTPUT.path)"
+                rm -f "${OUTPUT_FILE}.filtered" "${OUTPUT_FILE}.error"
+                exit 1
+            else
+                mv "${OUTPUT_FILE}.filtered" "$OUTPUT_FILE"
+                echo "Filtered results saved back to $OUTPUT_FILE"
+            fi
+        else
+            echo "KFP_GIT_URL is not set. Skipping false positive filtering."
+        fi
+
+        if [[ "$IMP_FINDINGS_ONLY" == "true" ]]; then
+            CSGREP_EVENT_FILTER='\[SC(1020|1035|1054|1066|1068|1073|1080|1083|1099|1113|1115|1127|1128|1143|2043|2050|'
+            CSGREP_EVENT_FILTER+='2055|2057|2066|2069|2071|2077|2078|2091|2092|2157|2171|2193|2194|2195|2215|2216|'
+            CSGREP_EVENT_FILTER+='2218|2224|2225|2242|2256|2258|2261)\]$'
+            csgrep --mode=json --event "$CSGREP_EVENT_FILTER" "$OUTPUT_FILE" > "$OUTPUT_FILE"
```

Comment:
This command cannot be safely executed because shell may overwrite `$OUTPUT_FILE` before its original content is read by `csgrep`.  I would suggest to move the whole `if` condition before the first invocation of `csgrep` and simply append the `--event=...` argument to the `CSGREP_OPTS` array in the `then` branch.  The current `--event='error|warning'` argument needs to be moved to the `else` branch.  This will also optimize out unnecessary serialization/deserialization of the JSON data (and speed up KFP processing in case the full list of findings is huge).

Reply:
addressed in https://github.com/konflux-ci/build-definitions/pull/1437/commits/ac33aad0a7fafa701249357663eda6d4aefda828